business/checkers/destination_rules/no_host_checker.go

```diff
@@ -1,33 +1,57 @@
 package destination_rules
 
 import (
+	"strconv"
+
 	"github.com/kiali/kiali/kubernetes"
 	"github.com/kiali/kiali/models"
 )
 
 type NoHostChecker struct {
 	Namespace       string
-	ServiceNames    []string
+	Services        map[string][]string
 	DestinationRule kubernetes.IstioObject
 }
 
-func (destinationRule NoHostChecker) Check() ([]*models.IstioCheck, bool) {
+// Check parses the DestinationRule definitions and verifies that they point to an existing service, including any subset definitions
```

Comment:
This is good. I wonder about the naming, as we check host/version == destination pair, I think no_destination_checker.go sounds more meaningful, wdyt ? but it's minor.